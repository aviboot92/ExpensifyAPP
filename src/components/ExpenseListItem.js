import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = (props) => (
    <div>
        <p>Descreption :{props.descreption}</p>
        <p>Amount: {props.amount}</p>
        <p>Created At: {props.createdAt}</p>
        <button
            id={props.id}
            onClick={(e) => {
            props.dispatch(removeExpense({id: e.target.id}));
        }}>Remove</button>
        <br/>
    </div>
);

export default connect()(ExpenseListItem);