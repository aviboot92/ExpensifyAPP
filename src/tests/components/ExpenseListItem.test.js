import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from './../../components/ExpenseListItem';

test('should render ExpenseListItem correctly', () => {
    const wrapper = shallow(<ExpenseListItem id="7" amount="1000" descreption="Coffee" createdAt= {10222}/>);
    expect(wrapper).toMatchSnapshot();
})