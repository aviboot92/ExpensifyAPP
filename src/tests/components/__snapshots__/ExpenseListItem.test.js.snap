// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div>
  <p>
    <Link
      to="/edit/7"
    >
       Descreption :
      Coffee
    </Link>
  </p>
  <p>
    Amount: 
    10
  </p>
  <p>
    Created At: 
    10222
  </p>
  <br />
</div>
`;
